src/EndcapTOF_geo.cpp

```diff
@@ -0,0 +1,721 @@
+    // << "\n\tdisk_zPos = " << disk_zPos
+    // << "\n\tdisk_det_height = " << disk_det_height
+    // << "\n\tdisk_det_height = " << disk_det_height
+    // << "\n\tcooling_tube_thickness = " << cooling_tube_thickness
+    // << "\n\tcooling_tube_diameter = " << cooling_tube_diameter
+    // << endl;
+
+    // create a solid for the beampipe cutout
+    Tube beampipe_cutout(0, disk_rMin,disk_det_height, 0, 2. * M_PI);
+    // create solids for the cooling/mountin disks (two Al disks)
+    Tube ttl_plate_1(0, disk_rMax,cooling_plate_height / 2., 0, 2. * M_PI);
+    Tube ttl_plate_2(0, disk_rMax,cooling_plate_height / 2., 0, 2. * M_PI);
+
+    // create the offset cutout in the disks
+    Position cutoutOffset(disk_xOffset, 0,0);
+    SubtractionSolid ttl_cooling_plate_1(ttl_plate_1, beampipe_cutout,cutoutOffset);
+    SubtractionSolid ttl_cooling_plate_2(ttl_plate_2, beampipe_cutout,cutoutOffset);
+
+    // set the material (Al)
+    Material disk_mat = description.material(diskdimension.materialStr());
+
+    // create the volumes and set visualization options
+    Volume ttl_cooling_plate_volume_1("ttl_cooling_plate_1", ttl_cooling_plate_1, disk_mat);
+    Volume ttl_cooling_plate_volume_2("ttl_cooling_plate_2", ttl_cooling_plate_2, disk_mat);
+    ttl_cooling_plate_volume_1.setVisAttributes(description.visAttributes("TOFAluminum"));
+    ttl_cooling_plate_volume_2.setVisAttributes(description.visAttributes("TOFAluminum"));
+
+    // place disks in assembly
+    pv = assembly.placeVolume(ttl_cooling_plate_volume_1, Position(0, 0, disk_zPos-cooling_tube_diameter/2.-cooling_plate_height/2.));
+    pv = assembly.placeVolume(ttl_cooling_plate_volume_2, Position(0, 0, disk_zPos+cooling_tube_diameter/2+cooling_plate_height/2.));
+
+    // ---------------------------------------
+    // Create Sensor Module Stack:
+    // ---------------------------------------
+    double sensor_width = 21.2 * mm;
+    double sensor_length = 42.0 * mm;
+    double baseplate_length = 43.1 * mm;
+    double baseplate_width = 56.5 * mm / 2;
+
+    const int nLayers = 8;
+    std::string strLayerName[nLayers] = {
+      "ThermalPad","ALN","LairdFilm","ROC","Solder","Sensor","Epoxy","AIN"
+      };
+    Material materialLayer[nLayers] = {
+      description.material("Graphite"), description.material("AluminumNitrate"), description.material("Graphite"), description.material("Plexiglass"), description.material("Tin"), description.material("Silicon"), description.material("Epoxy"), description.material("AluminumNitrate")
+    };
+    double thicknessLayer[nLayers] = {
+        0.25 * mm, 0.79 * mm, 0.08 * mm, 0.25 * mm, 0.03 * mm, 0.3 * mm, 0.08 * mm, 0.51 * mm};
+    double widthLayer[nLayers] = {
+        baseplate_width - 0.3 * mm,
+        baseplate_width,
+        sensor_width + 1 * mm,
+        sensor_width + 1 * mm,
+        sensor_width - 0.2 * mm,
+        sensor_width,
+        sensor_width,
+        baseplate_width - 4 * mm};
```

Comment:
There is still a lot of parametrization going on inside the cpp plugin. This would better be moved to the xml file with a `layer` support (as other detectors are doing). That also guarantees consistency since each layer would be required to have a thickenss, material, and name.

Reply:
Moved these definitions to the xml file.